mongo/document/document.go

```diff
@@ -46,64 +48,85 @@ type Closer interface {
 	Close()
 }
 
+var ErrValidateNoInternalErrorFunc = stderrors.New("no validation returning an internal error has been implemented")
+
 type Validable interface {
-	Validate(ctx context.Context) (*errors.ValidationErrors, error)
+	// Validate will be used if no ValidateWithInternalError is defined on a document
+	// It is not useful to have both defined on a document, only ValidationWithInternalError
+	// would be used in this case
+	Validate(ctx context.Context) *errors.ValidationErrors
```

Comment:
question: do we want to deprecate this method? In such case, we should document it IMO:

```suggestion
	// Deprecated: ...
	Validate(ctx context.Context) *errors.ValidationErrors
```

Reply:
I don't think we want to deprecate, it's just 2 ways to define validation, I think that keeping a simple signature for the comon case works well.